test/e2e/tkn_builder.go

```diff
@@ -0,0 +1,37 @@
+package e2e
+
+import (
+	"log"
+	"testing"
+
+	"gotest.tools/icmd"
+)
+
+type TknRunner struct {
+	path string
+}
+
+func NewTknRunner(tknPath string) TknRunner {
+	return TknRunner{
+		path: tknPath,
+	}
+}
+
+func (tkn *TknRunner) BuildTknClient() string {
```

Comment:
I am not sure why we build the tkn binary in there. That could be a "requirement" for running those tests, couldn't it ?

Reply:
I am building this to run all our e2e tests inside a temp file and gets removed once e2e tests finishes of its execution

Reply:
Right, but this mean we need to have the source and go to run thoses tests.
Usually I prefer making the assumption the `cli` got build before the test runs (and we have a default where we look for the built cli, and it could be specified when running the e2e)

Reply:
[This](https://github.com/moby/moby/blob/09226c4442716b4f057c0260fd86dda078d433ed/integration-cli/environment/environment.go#L10) is how we did it for `moby/moby`